### PR TITLE
Admin users

### PR DESCRIPTION
**Pull request by AlonaDorozhy**
### **build**
build

### **lint**
lint

### **test**
test

### CODE REVIEW

File: src/app/admin-users/helpers/helper-details/helper-details.component.html [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                    </mat-form-field>
+                    <mat-form-field class="details-full-width">
+                        <input matInput placeholder="Birthday" formControlName="birthday">
+                    </mat-form-field>
+                    <mat-form-field class="details-full-width">
+                        <input matInput placeholder="Rating" formControlName="rating">
+                    </mat-form-field>
+                </mat-card-title>
+                <mat-card-actions>
+                    <button class="edit-wrap" mat-button (click)="onEdit()" [disabled]="!isEdiDisabled">Edit</button>
```
Comment (vshovkuta) [inline src/app/admin-users/helpers/helper-details/helper-details.component.html]: Please rename property 'isEdiDisabled' from class 'HelperDetailsComponent' (and other places) in 'isEdi**t**Disabled'. Or is it correct name?

File: src/app/admin-users/managers/manager-details/manager-details.component.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                        <button class="edit-wrap" mat-button (click)="onEdit()" [disabled]="!isEdiDisabled">Edit</button>
+                    </mat-card-actions>
+                </form>
+            </mat-card-content>
+            <mat-card-actions *ngIf=" visibleFields" class="button">
+                <button mat-raised-button class="reset" (click)="onReset()" [disabled]="isEdiDisabled"><mat-icon>settings_backup_restore</mat-icon>Reset</button>
+                <button mat-raised-button class="delete" [routerLink]="['/manager', manager.id]" (click)="deleteUser()"
+                    [disabled]="isEdiDisabled || profileForm.invalid"><mat-icon>delete_forever</mat-icon> Delete user</button>
+                <button mat-raised-button class="saveBtn" [routerLink]="['/manager', manager.id]" (click)="changeInfo()"
+                    [disabled]="isEdiDisabled || profileForm.invalid"><mat-icon>done_all</mat-icon>Save changes</button>
```
Comment (vshovkuta) [inline src/app/admin-users/managers/manager-details/manager-details.component.html]: Please enter 1 space before the name of the buttons 
`</mat-icon>Save changes</button> => </mat-icon>·Save changes</button>`
or 
create a new style with padding-right/margin-right for tags <mat-icon>

File: src/app/admin-users/managers/manager-item/manager-item.component.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<div>
+    <mat-card class="user-card">
+        <mat-card-header>
+            <div mat-card-avatar class="example-header-image"></div>
+            <mat-card-title class="title">{{manager.name}} {{manager.patronymic}} {{manager. surname}}
+            </mat-card-title>
+        </mat-card-header>
+        <mat-card-content>
+            <p><img mat-card-image class="avatar" src="{{manager.photoPath}}" alt="Photo of a Shiba Inu"></p>
+            <p>Birthday: {{manager.birthday| date: 'dd/MM/yyyy'}}</p>
```
Comment (vshovkuta) [inline src/app/admin-users/managers/manager-item/manager-item.component.html]: `{{manager.birthday·| date`

File: src/app/admin-users/managers/manager-item/manager-item.component.scss [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    font-size: 18px;
+    height: 35px;
+}
+
+.avatar {
+    width: 100px;
+    float: left;
+    margin: 10px;
+}
+.button{
```
Comment (vshovkuta) [inline src/app/admin-users/managers/manager-item/manager-item.component.scss]: Please enter 1 free line between styles

File: src/app/admin-users/managers/manager-details/manager-details.component.html (hunk 2 above)
Comment (vshovkuta) [reply src/app/admin-users/managers/manager-details/manager-details.component.html]: Also, you can align the icons and the captions in one line 
alona_dev2

File: src/app/admin-users/managers/manager-item/manager-item.component.html (hunk 3 above)
Comment (AlonaDorozhy) [reply src/app/admin-users/managers/manager-item/manager-item.component.html]: I don't understand, wats wrong?

File: src/app/admin-users/managers/manager-item/manager-item.component.scss (hunk 4 above)
Comment (AlonaDorozhy) [reply src/app/admin-users/managers/manager-item/manager-item.component.scss]: added

File: src/app/admin-users/managers/manager-item/manager-item.component.html (hunk 3 above)
Comment (vshovkuta) [reply src/app/admin-users/managers/manager-item/manager-item.component.html]: space before `| date`
=)

File: src/app/admin-users/managers/manager-item/manager-item.component.scss (hunk 4 above)
Comment (vshovkuta) [reply src/app/admin-users/managers/manager-item/manager-item.component.scss]: And necessary space between `selector` & `{`, I've forgotten to say this in first message, sorry)

File: src/app/admin-users/managers/manager-details/manager-details.component.html (hunk 2 above)
Comment (AlonaDorozhy) [reply src/app/admin-users/managers/manager-details/manager-details.component.html]: image

File: src/app/admin-users/admin-users.component.ts [hunk 5]
```diff
@@ -1,28 +1,37 @@
...
-  public users$: Observable<Helper[]>;
-  public managers$: Observable<Manager[]>;
 
-  constructor(private helperService: HelpersService, private managerService: ManagersService) { }
-
-  ngOnInit(): void {
-    this.users$ = this.helperService.getUserInfo();
-    this.managers$ = this.managerService.getManagers();
+  navLinks: any[];
+  activeLinkIndex = -1;
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.ts]: add visibility modifiers

File: src/app/admin-users/admin-users.component.ts [hunk 6]
```diff
@@ -1,28 +1,37 @@
...
 export class AdminUsersComponent implements OnInit {
-  public users$: Observable<Helper[]>;
-  public managers$: Observable<Manager[]>;
 
-  constructor(private helperService: HelpersService, private managerService: ManagersService) { }
-
-  ngOnInit(): void {
-    this.users$ = this.helperService.getUserInfo();
-    this.managers$ = this.managerService.getManagers();
+  navLinks: any[];
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.ts]: do not use any as a type

File: src/app/admin-users/admin-users.component.ts [hunk 7]
```diff
@@ -1,28 +1,37 @@
...
+  activeLinkIndex = -1;
+  constructor(private router: Router) {
+    this.navLinks = [
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.ts]: 1) why do you need to assign it in constructor? the same will be if you assign it where the property was defined;
2) why do you need such data in component? this looks like a part of router logic

File: src/app/admin-users/admin-users.component.ts [hunk 8]
```diff
@@ -1,28 +1,37 @@
...
+      },
+    ];
   }
-  public usertrackBy(index: number, user: Helper): string {
-    return user.id;
-  }
-  public managertrackBy(index: number, manager: Manager): string {
-    return manager.id;
+  ngOnInit(): void {
+    this.router.events.subscribe((res) => {
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.ts]: unsubscribe

File: src/app/admin-users/helpers/helper-details/helper-details.component.html [hunk 9]
```diff
@@ -0,0 +1,45 @@
+<div>
+        <button mat-raised-button color="primary"><mat-icon>reply</mat-icon> Back to helpers </button>
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-details/helper-details.component.html]: fix formatting

File: src/app/admin-users/helpers/helper-details/helper-details.component.html [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+                    <mat-form-field class="details-full-width">
+                        <input matInput placeholder="Rating" formControlName="rating">
+                    </mat-form-field><br>
+                </mat-card-title>
+                <mat-card-actions>
+                    <button class="editBtn" mat-button (click)="onEdit()" [disabled]="!isEditDisabled">Edit</button>
+                </mat-card-actions>
+            </form>
+        </mat-card-content>
+        <mat-card-actions *ngIf=" visibleFields" class="button">
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-details/helper-details.component.html]: remove extra space from ngIf

File: src/app/admin-users/helpers/helper-details/helper-details.component.html [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+                    </mat-form-field><br>
+                    <mat-form-field class="details-full-width">
+                        <input matInput placeholder="Birthday" formControlName="birthday">
+                    </mat-form-field><br>
+                    <mat-form-field class="details-full-width">
+                        <input matInput placeholder="Rating" formControlName="rating">
+                    </mat-form-field><br>
+                </mat-card-title>
+                <mat-card-actions>
+                    <button class="editBtn" mat-button (click)="onEdit()" [disabled]="!isEditDisabled">Edit</button>
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-details/helper-details.component.html]: 1) class `editBtn`? do you still need this custom classes?
2) be consistent and use kebab-case for styles class names (ie `edit-btn`)

File: src/app/admin-users/helpers/helper-details/helper-details.component.html [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+                </mat-card-actions>
+            </form>
+        </mat-card-content>
+        <mat-card-actions *ngIf=" visibleFields" class="button">
+            <button mat-raised-button class="reset"  (click)="onReset()" [disabled]="isEditDisabled"><mat-icon>settings_backup_restore</mat-icon>Reset</button>
+            <button mat-raised-button class="delete" [routerLink]="['/helper', helper.id]" (click)="deleteUser()"
+                [disabled]="isEditDisabled || profileForm.invalid"><mat-icon>delete_forever</mat-icon>Delete user</button>
+            <button mat-raised-button class="saveBtn" [routerLink]="['/helper', helper.id]" (click)="changeInfo()"
+                [disabled]="isEditDisabled || profileForm.invalid"><mat-icon>done_all</mat-icon>Save changes</button>
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-details/helper-details.component.html]: you need to decide if you want to apply routerLink to these buttons or you own click handlers

File: src/app/admin-users/helpers/helper-details/helper-details.component.scss [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+.heder-card {
+    text-align: center;
+}
+
+.mat-input-element.mat-form-field-autofill-control.cdk-text-field-autofill-monitored.ng-untouched.ng-pristine {  
+    color: black;
+}
+
+.mat-input-element.mat-form-field-autofill-control.cdk-text-field-autofill-monitored.ng-untouched.ng-pristine.ng-valid{
+    color: rgb(44, 87, 206);
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-details/helper-details.component.scss]: move all colors to variables

File: src/app/admin-users/helpers/helper-details/helper-details.component.scss [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+
+.heder-card {
+    text-align: center;
+}
+
+.mat-input-element.mat-form-field-autofill-control.cdk-text-field-autofill-monitored.ng-untouched.ng-pristine {  
+    color: black;
+}
+
+.mat-input-element.mat-form-field-autofill-control.cdk-text-field-autofill-monitored.ng-untouched.ng-pristine.ng-valid{
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-details/helper-details.component.scss]: you sure you  need this selector for it? please use scss feature to organize this styles in one block so it is clear what you need them for

File: src/app/admin-users/helpers/helper-details/helper-details.component.scss [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+    color: rgb(35, 97, 73);
+}
+
+.reset{
+    background-color: rgb(231, 144, 44);
+    color : rgb(255, 255, 255);
+    margin: 0 8px;
+}
+
+.delete{
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-details/helper-details.component.scss]: please check formatting in this file: you have some spaces missing and some extra ones too

File: src/app/admin-users/helpers/helper-details/helper-details.component.ts [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+    this.visibleFields = !this.visibleFields;
+  }
+
+  public onReset(): void {
+    this.patchFormValues(this.helper);
+  }
+  public deleteUser() {
+    this.helpersService
+      .deleteUser(this.helperId)
+      .subscribe();
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-details/helper-details.component.ts]: if you need navigation to other route after your click handlers you need to do it in this subscribe handlers (so it is occurred after your requests), and the same for the method below

File: src/app/admin-users/helpers/helper-item/helper-item.component.scss [hunk 17]
```diff
@@ -0,0 +1,20 @@
+.user-card {
+    max-width: 370px;
+    height: 200px;
+    margin: 0 10px;
+}
+
+.title{
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-item/helper-item.component.scss]: missing space here and on line 18

File: src/app/admin-users/helpers/helper-item/helper-item.component.ts [hunk 18]
```diff
@@ -0,0 +1,19 @@
...
+  selector: 'app-helper-item',
+  templateUrl: './helper-item.component.html',
+  styleUrls: ['./helper-item.component.scss']
+})
+export class HelperItemComponent implements OnInit {
+  public helperId: number;
+  @Input() helper: Helper;
+  constructor() { }
+  ngOnInit() {
+  }
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-item/helper-item.component.ts]: remove constructor and ngOnInit hook if you don't need them

File: src/app/admin-users/helpers/helper-item/helper-item.component.ts [hunk 19]
```diff
@@ -0,0 +1,19 @@
...
+
+  public selecHelper(helper: { id: number; }) {
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-item/helper-item.component.ts]: helper has the type Helper, why do you set it to something else here?

File: src/app/admin-users/helpers/helper-item/helper-item.component.ts [hunk 20]
```diff
@@ -0,0 +1,19 @@
...
+    this.helperId = helper.id;
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-item/helper-item.component.ts]: why do you need it? this component will always have just 1 helper object and you can have it id from there

File: src/app/admin-users/helpers/helper-item/helper-item.component.html [hunk 21]
```diff
@@ -0,0 +1,19 @@
...
+        </mat-card-header>
+        <mat-card-content>
+            <p><img mat-card-image class="avatar" src="{{helper.photoPath}}" alt="Photo of a Shiba Inu"></p>
+            <p>Birthday: {{helper.birthday | date:'dd/MM/yyyy'}}</p>
+            <p>Rating: {{helper.rating}}</p>
+        </mat-card-content>
+        <mat-card-actions class="button">
+                <button mat-raised-button color="primary" 
+                [routerLink]="['/helper', helper.id]"
+                (click)=" selecHelper(helper)"> More</button>
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-item/helper-item.component.html]: why do you need this click handler? you already routerLink here

File: src/app/admin-users/helpers/helper-item/helper-item.component.scss (hunk 17 above)
Comment (ksenya-ta) [reply src/app/admin-users/helpers/helper-item/helper-item.component.scss]: and the same issues in other styles files, please check them all

File: src/app/admin-users/helpers/helpers-list/helpers-list.component.html [hunk 22]
```diff
@@ -0,0 +1,12 @@
+<div class="side">
+    <mat-grid-list cols="5">
+        <mat-grid-tile-header>
+            <button class="addPerson" mat-raised-button color="primary">
+                <mat-icon>person_add</mat-icon>Add Helper
+            </button>
+        </mat-grid-tile-header>
+        <mat-grid-tile *ngFor="let helper of helpers$ | async; trackBy: TrackByHelpers">
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helpers-list/helpers-list.component.html]: the method is called `trackByHelpers` not `TrackByHelpers`

File: src/app/admin-users/helpers/services/helper.service.ts [hunk 23]
```diff
@@ -1,17 +1,17 @@
...
 import { HttpClient } from '@angular/common/http';
-import { Helper } from './helper.models';
+import { Helper } from '../models/helper.model';
 import { Observable } from 'rxjs';
 @Injectable({
   providedIn: 'root'
 })
 export class HelpersService {
-  apiUsers = 'https://cors-anywhere.herokuapp.com/https://familynetserver.azurewebsites.net/api/v1/users/';
+  apiUsers = 'https://familynetserver.azurewebsites.net/api/v1/volunteers/';
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/services/helper.service.ts]: don't you need to use the config to get this URL like it was implemented in other services?

File: src/app/admin-users/helpers/services/helper.service.ts [hunk 24]
```diff
@@ -1,17 +1,17 @@
...
   constructor(private httpClient: HttpClient) { }
-  public getAllUsers(url?: string): Observable<Helper[]> {
+  public getAllUsers(): Observable<Helper[]> {
     return this.httpClient.get<Helper[]>(`${this.apiUsers}`);
   }
-  public getUserById(id: string): Observable<any> {
+  public getHelperById(id: string): Observable<any> {
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/services/helper.service.ts]: do not use any as a type

File: src/app/admin-users/managers/manager-details/manager-details.component.scss [hunk 25]
```diff
@@ -0,0 +1,78 @@
...
+
+.mat-card-actions {
+    text-align: center;
+}
+
+.heder-card {
+    text-align: center;
+}
+
+.mat-input-element.mat-form-field-autofill-control.cdk-text-field-autofill-monitored.ng-untouched.ng-pristine {
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/manager-details/manager-details.component.scss]: do not copy the styles from other components, if you need the same styles make it common so you can include it to both components or create a mixin for something you need and just use it

File: src/app/admin-users/managers/manager-details/manager-details.component.ts [hunk 26]
```diff
@@ -0,0 +1,98 @@
...
+    this.createForm();
+    this.toggleForm();
+    this.managerId = this.activatedRoute.snapshot.params.id;
+    this.managersService
+      .getManagerById(this.managerId)
+      .pipe(takeUntil(this.unsubscribe))
+      .subscribe(managers => {
+        this.patchFormValues(managers);
+        this.manager = managers;
+        return this.manager;
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/manager-details/manager-details.component.ts]: why do you need this return?

File: src/app/admin-users/managers/manager-details/manager-details.component.ts [hunk 27]
```diff
@@ -0,0 +1,98 @@
...
+      });
+  }
+
+  private createForm(): void {
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/manager-details/manager-details.component.ts]: private methods should be defined after all public and protected ones

File: src/app/admin-users/managers/manager-item/manager-item.component.html [hunk 28]
```diff
@@ -0,0 +1,19 @@
+<div>
+    <mat-card class="user-card">
+        <mat-card-header>
+            <div mat-card-avatar class="example-header-image"></div>
+            <mat-card-title class="title">{{manager.name}} {{manager.patronymic}} {{manager. surname}}
+            </mat-card-title>
+        </mat-card-header>
+        <mat-card-content>
+            <p><img mat-card-image class="avatar" src="{{manager.photoPath}}" alt="Photo of a Shiba Inu"></p>
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/manager-item/manager-item.component.html]: Photo of a Shiba Inu? or maybe photo of  manager's full name?

File: src/app/admin-users/managers/manager-item/manager-item.component.ts [hunk 29]
```diff
@@ -0,0 +1,20 @@
...
+  selector: 'app-manager-item',
+  templateUrl: './manager-item.component.html',
+  styleUrls: ['./manager-item.component.scss']
+})
+export class ManagerItemComponent implements OnInit {
+  public managerId: number;
+  @Input() manager: Manager;
+  constructor() { }
+  ngOnInit() {
+  }
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/manager-item/manager-item.component.ts]: same here - remove not needed methods/interfaces

File: src/app/admin-users/managers/managers-list/managers-list.component.ts [hunk 30]
```diff
@@ -0,0 +1,20 @@
...
+  styleUrls: ['./managers-list.component.scss']
+})
+export class ManagersListComponent implements OnInit {
+  public managers$: Observable<Manager[]>;
+  constructor(private managerService: ManagersService) { }
+
+  ngOnInit(): void {
+    this.managers$ = this.managerService.getManagers();
+  }
+  public TrackByManagers(index: number, user: Manager): string {
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/managers-list/managers-list.component.ts]: name it in camelCase

File: src/app/admin-users/managers/manager-details/manager-details.component.ts (hunk 27 above)
Comment (AlonaDorozhy) [reply src/app/admin-users/managers/manager-details/manager-details.component.ts]: This is modified, for some unknown reason it does not appear here.
https://github.com/vderunov/shelter/blob/Admin-Users/src/app/admin-users/managers/manager-details/manager-details.component.ts

File: src/app/admin-users/helpers/helper-item/helper-item.component.html [hunk 31]
```diff
@@ -0,0 +1,18 @@
+<div>
+    <mat-card class="user-card">
+        <mat-card-header>
+            <mat-card-title class="title">{{ helper.name }} {{ helper.patronymic }} {{ helper.surname }}
+            </mat-card-title>
+        </mat-card-header>
+        <mat-card-content>
+            <p><img mat-card-image class="avatar" src="{{ helper.photoPath }}" alt="Photo of {{ helper.name }} {{ helper.surname }}"></p>
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-item/helper-item.component.html]: same issue with bindings

File: src/app/admin-users/helpers/helper-item/helper-item.component.ts [hunk 32]
```diff
@@ -0,0 +1,18 @@
...
+  selector: 'app-helper-item',
+  templateUrl: './helper-item.component.html',
+  styleUrls: ['./helper-item.component.scss']
+})
+export class HelperItemComponent {
+
+  @Input() helper: Helper;
+  constructor(private router: Router) { }
+
+  public selectHelper(helper: Helper) {
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helper-item/helper-item.component.ts]: why do you create additional method call if you can use the routerLink directive for it?

File: src/app/admin-users/helpers/services/helper.service.ts [hunk 33]
```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Helper } from '../models/helper.model';
+import { Observable } from 'rxjs';
+import { concatMap } from 'rxjs/operators';
+import { ConfigService } from 'src/app/shared/services/config/config.service';
+import { Config } from 'src/app/shared/services/config/config.interface';
+@Injectable({
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/services/helper.service.ts]: add new line after imports

File: src/app/admin-users/managers/manager-details/manager-details.component.html [hunk 34]
```diff
@@ -0,0 +1,59 @@
...
+                    <button  mat-button (click)="onEdit()"
+                     [disabled]="!isEditDisabled">Edit</button>
+                </mat-card-actions>
+            </form>
+        </mat-card-content>
+        <mat-card-actions *ngIf="visibleFields" class="button">
+            <button mat-raised-button class="reset" (click)="onReset()"
+             [disabled]="isEditDisabled">
+                <mat-icon>settings_backup_restore</mat-icon>Reset
+
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/manager-details/manager-details.component.html]: why do you need an extra new line here?

File: src/app/admin-users/managers/manager-details/manager-details.component.ts [hunk 35]
```diff
@@ -0,0 +1,100 @@
...
+  public deleteUser() {
+    this.router.navigate(['users']);
+  }
+
+  public changeInfo() {
+    this.managersService
+      .updateManager(this.profileForm.value, this.managerId)
+      .subscribe();
+    this.router.navigate(['/managers', this.manager.id]);
+    this.onEdit();
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/manager-details/manager-details.component.ts]: 1) decide what you need here: onEdit call or navigation
2) API calls may take some time so you need to do anything you need after it in the subscribe

File: src/app/admin-users/managers/manager-item/manager-item.component.ts [hunk 36]
```diff
@@ -0,0 +1,20 @@
...
+  templateUrl: './manager-item.component.html',
+  styleUrls: ['./manager-item.component.scss']
+})
+
+export class ManagerItemComponent {
+  @Input() manager: Manager;
+
+  constructor(private router: Router) { }
+
+  public selecManager(manager: Manager) {
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/manager-item/manager-item.component.ts]: use routerLink directive instead

File: src/app/admin-users/managers/services/manager.service.ts [hunk 37]
```diff
@@ -0,0 +1,44 @@
+import { Injectable, OnInit } from '@angular/core';
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Manager } from '../models/manager.model';
+import { Observable } from 'rxjs';
+import { Config } from 'src/app/shared/services/config/config.interface';
+import { ConfigService } from 'src/app/shared/services/config/config.service';
+import { concatMap } from 'rxjs/operators';
+@Injectable({
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/services/manager.service.ts]: fix formatting in this file

File: src/app/admin-users/managers/services/manager.service.ts [hunk 38]
```diff
@@ -0,0 +1,44 @@
...
+    providedIn: 'root'
+})
+
+export class ManagersService {
+   public manager: Manager;
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/services/manager.service.ts]: all services should be stateless, do not save any data on it

File: src/app/admin-users/managers/services/manager.service.ts [hunk 39]
```diff
@@ -0,0 +1,44 @@
...
+        return this.http.get<Manager>(`${api}/${id}`);
+    }
+
+    public updateManager(api, id: string) {
+        return this.http.put(`${api}${id}`, this.manager);
+    }
+
+    public deleteManager(api, id: string) {
+        return this.http.delete(`${api}${id}`);
+    }
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/services/manager.service.ts]: 1) add types for all properties and returns
2) and how the api variable is supposed to be passed to these methods?

File: src/app/app-routing.module.ts [hunk 40]
```diff
@@ -3,11 +3,13 @@ import { Routes, RouterModule } from '@angular/router';
...
 import { RegistrationUserComponent } from './registration-user/registration-user.component';
 import { AdminUsersComponent } from './admin-users/admin-users.component';
+import { HelpersListComponent } from './admin-users/helpers/helpers-list/helpers-list.component';
+import { ManagersListComponent } from './admin-users/managers/managers-list/managers-list.component';
 
 const routes: Routes = [
   { path: 'login', component: LoginComponent },
   { path: 'registraction-user', component: RegistrationUserComponent },
-  { path: 'users', component: AdminUsersComponent }
+  { path: 'users/', component: AdminUsersComponent },
```
Comment (ksenya-ta) [inline src/app/app-routing.module.ts]: why do you need extra / and ,

File: src/app/shared/style/variables.scss [hunk 41]
```diff
@@ -4,3 +4,9 @@ $colorRedForError: #FF0000;
 $button-primary-background-color: aliceblue;
 $selected-background-color: #CFD8DC;
 $selected-hover-background-color: #BBD8DC;
+$color-black: #000000;
```
Comment (ksenya-ta) [inline src/app/shared/style/variables.scss]: make sure the names for variables are consistent

File: src/app/admin-users/managers/services/manager.service.ts [hunk 42]
```diff
@@ -0,0 +1,44 @@
...
+
+    public getManagerById(id: string = ''): Observable<Manager> {
+        return this.configService.getConfig().pipe(
+            concatMap((config: Config) => this.getManager(config.managersApi, id)
+            )
+        );
+    }
+
+    public getManager(api, id): Observable<Manager> {
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/services/manager.service.ts]: btw types are still missing

Comment (AlonaDorozhy) [reply src/app/admin-users/managers/services/manager.service.ts]: https://github.com/vderunov/shelter/blob/Admin-Users/src/app/admin-users/managers/services/manager.service.ts

It looks different, not updated

**Pull request merged**